pkg/helm/urlfetcher.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archiver"
+
+	"istio.io/pkg/log"
+)
+
+// FileDownloader is wrapper of HTTP client to download files
+type FileDownloader struct {
+	// client is a HTTP/HTTPS client.
+	client *http.Client
+}
+
+// URLFetcher is used to fetch and manipulate charts from remote url
+type URLFetcher struct {
+	// url is url to download the charts
+	url string
+	// verifyURL is url to download the verification file
+	verifyURL string
```

Comment:
This may also be supplied from the caller through the API (from IstioControlPlane CR) so we should add that as an option, maybe when the fetcher is created.

Reply:
Can you clarify this point?

Reply:
I don't see a public API, only private, so how would a caller create a fetcher for a particular URL?